### PR TITLE
snap-{seccomp,confine}: replace global seccomp filter with template

### PR DESCRIPTION
**Pull request by zyga**
The global.bin seccomp filter was written before we had support for explicit deny rules in snap-seccomp. With these rules we can replace the hard to followup logic of the global filter and just make the rules part of the standard seccomp template.

The global rules are best summarized in this comment:
```
struct scmp_arg_cmp no_tty_inject = {
    /* We learned that existing programs make legitimate requests with all
     * bits set in the more significant 32bit word of the 64 bit double
     * word. While this kernel behavior remains suspect and presumably
     * undesired it is unlikely to change for backwards compatibility
     * reasons. As such we cannot block all requests with high-bits set.
     *
     * When faced with ioctl(fd, request); refuse to proceed when
     * request&0xffffffff == TIOCSTI. This specific way to encode the
     * filter has the following important properties:
     *
     * - it blocks ioctl(fd, TIOCSTI, ptr).
     * - it also blocks ioctl(fd, (1UL<<32) | TIOCSTI, ptr).
     * - it doesn't block ioctl(fd, (1UL<<32) | (request not equal to TIOCSTI), ptr); */
    .arg = 1,
    .op = SCMP_CMP_MASKED_EQ,
    .datum_a = 0xffffffffUL,
    .datum_b = TIOCSTI,
};
sc_err = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EPERM), sys_ioctl_nr, 1, no_tty_inject);
```
and the same for `TIOCLINUX`.

### CODE REVIEW

**Review by alexmurray — APPROVED** [review]
LGTM - the changes to the lp-1812973 regression test to change the expected errno to EACCESS match the explicit denial value in the seccomp filter, demonstrating that the seccomp filter is appropriately denying these accesses as expected.

**Review by jslarraz — APPROVED** [review]
LGTM, Thanks @zyga

File: tests/regression/lp-1812973/lp-1812973.c [hunk 1]
```diff
@@ -29,15 +29,15 @@ int main(int argc, char **argv) {
     if (argc == 2 && strcmp(argv[1], "--evil") == 0) {
         res = ioctl64(fd, TIOCSTI, (unsigned long)&pushmeback);
         saved_errno = errno;
-        printf("normal TIOCSTI: %d (%m) (expect EPERM)\n", res);
-        if (res < 0 && saved_errno == EPERM) {
+        printf("normal TIOCSTI: %d (%m) (expect EACCES)\n", res);
```
Comment (bboozzoo) [inline tests/regression/lp-1812973/lp-1812973.c:32]: perhaps add a comment:
```
// denial is explicit so we're expecting EACCESS instead of EPERM
```

File: cmd/snap-seccomp-blacklist/snap-seccomp-blacklist.c
Comment (bboozzoo) [inline cmd/snap-seccomp-blacklist/snap-seccomp-blacklist.c:1]: hm interfaces/seccomp/backend.go should be updated too? see Backend.Initialize()

Comment (zyga) [reply cmd/snap-seccomp-blacklist/snap-seccomp-blacklist.c:1]: Done, thanks for spotting that.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: interfaces/seccomp/backend.go [hunk 2]
```diff
@@ -79,63 +78,12 @@ type Backend struct {
...
-	} else {
-		globalProfile = globalProfileLE
-	}
-	content := map[string]osutil.FileState{
-		fname: &osutil.MemoryFileState{Content: globalProfile, Mode: 0644},
-	}
-	if err := os.MkdirAll(dir, 0755); err != nil {
-		return fmt.Errorf("cannot create directory for seccomp profiles %q: %s", dir, err)
-	}
-	_, _, err := osutil.EnsureDirState(dir, fname, content)
```
Comment (bboozzoo) [inline interfaces/seccomp/backend.go:134]: FWIW it's be nice to clean up the file, but there's no guarantee we're not racing with snap apps/services that could be starting up.  Maybe we need a follwup with some sort of ensure handler that effectively removes this file.

Comment (zyga) [reply interfaces/seccomp/backend.go:134]: Yes I thought about removing the file but I think it's really tricky and probably not worth it. I wonder how to track it so that we don't just forget?
